core-api/src/main/java/com/optimizely/ab/bucketing/Bucketer.java

```diff
@@ -102,6 +109,19 @@ private Variation bucketToVariation(@Nonnull Experiment experiment,
                                         @Nonnull String userId) {
         // "salt" the bucket id using the experiment id
         String combinedBucketId = userId + experiment.getId();
+
+        // If a persistent bucketer instance is present then check it for a persisted variation
+        if (persistentBucketer != null) {
+            Variation variation = persistentBucketer.restoreActivation(projectConfig, userId, experiment);
+            if (variation != null) {
+                logger.info("Returning previously activated variation \"{}\" from persistent bucketer.", variation.getKey());
+                // A variation is stored for this combined bucket id
+                return variation;
```

Comment:
I think it's up to the persistent bucketing strategy implementation.  Users will need to use `ProjectConfig` to look up the variation instance that matches the id they saved.  If it doesn't exist they will return null.  

From `AndroidPersistentBucketer` 

```